com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/helpers/ApplicationColorDetector.java

```diff
@@ -1,29 +1,52 @@
 package com.sap.cloud.lm.sl.cf.core.helpers;
 
+import java.util.Arrays;
 import java.util.Date;
+import java.util.List;
 
+import org.flowable.variable.api.history.HistoricVariableInstance;
+
+import com.sap.cloud.lm.sl.cf.core.Constants;
+import com.sap.cloud.lm.sl.cf.core.dao.OperationDao;
+import com.sap.cloud.lm.sl.cf.core.flowable.FlowableFacade;
 import com.sap.cloud.lm.sl.cf.core.message.Messages;
 import com.sap.cloud.lm.sl.cf.core.model.ApplicationColor;
 import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
 import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.Phase;
+import com.sap.cloud.lm.sl.cf.web.api.model.Operation;
+import com.sap.cloud.lm.sl.cf.web.api.model.State;
 import com.sap.cloud.lm.sl.common.ConflictException;
 
 public class ApplicationColorDetector {
 
     private static final ApplicationColor COLOR_OF_APPLICATIONS_WITHOUT_SUFFIX = ApplicationColor.BLUE;
 
-    public ApplicationColor detectFirstDeployedApplicationColor(DeployedMta deployedMta) {
+    public ApplicationColor detectLiveApplicationColor(DeployedMta deployedMta, OperationDao operationDao, String correlationId,
+        FlowableFacade flowableFacade) {
         if (deployedMta == null) {
             return null;
         }
-        ApplicationColor firstApplicationColor = null;
-        Date firstApplicationColorCreatedOn = null;
-        for (DeployedMtaModule module : deployedMta.getModules()) {
-            Date moduleCreatedOn = module.getCreatedOn();
-            if (firstApplicationColorCreatedOn == null || firstApplicationColorCreatedOn.after(moduleCreatedOn)) {
-                firstApplicationColorCreatedOn = moduleCreatedOn;
-                firstApplicationColor = getApplicationColor(module);
-            }
+        ApplicationColor firstApplicationColor = getFirstApplicationColor(deployedMta);
+
+        Operation currentOperation = operationDao.find(correlationId);
+        if (currentOperation == null) {
+            return firstApplicationColor;
+        }
+        Operation lastOperation = operationDao.findFirstByMtaIdAndEndedAtIsNotNullOrderByEndedAtDesc(currentOperation.getMtaId());
+        if (lastOperation == null || lastOperation.getState() != State.ABORTED) {
+            return firstApplicationColor;
+        }
+        List<HistoricVariableInstance> xs2HistoricInstanceVariables = getXs2HistoricInstanceVariables(flowableFacade, lastOperation);
+        String lastDeployedMtaColorValue = flowableFacade.getVariableFromHistoricVariables(xs2HistoricInstanceVariables,
+            Constants.MTA_COLOR);
+        String phase = flowableFacade.getVariableFromHistoricVariables(xs2HistoricInstanceVariables, Constants.PHASE);
+
+        if (lastDeployedMtaColorValue == null || phase == null) {
+            return firstApplicationColor;
+        }
+        if (ApplicationColor.valueOf(lastDeployedMtaColorValue) != firstApplicationColor && Phase.valueOf(phase) == Phase.UNDEPLOY) {
```

Comment:
Simplify this:
- If the phase of the previous process is undeploy then:
  - Return the `lastDeployedMtaColor.getAlternativeColor()`.